storage/store.go

```diff
@@ -1701,6 +1707,42 @@ func (s *Store) MergeRange(subsumingRng *Replica, updatedEndKey roachpb.RKey, su
 	return subsumingRng.setDesc(&copy)
 }
 
+// If the subsuming replica has the range lease, then we update its timestamp
+// cache with the entries from the subsumed. If it's not the lease holder,
+// then the timestamp cache doesn't matter (in fact it should be empty, to
+// save memory).
+func (s *Store) maybeMergeTimestampCaches(
+	subsumingRng *Replica,
+	subsumedRng *Replica,
```

Comment:
`subsuming` and `subsumed` are too similar for my tastes. I think we should use the terminology from splits: `LHS` and `RHS` (and `leftRng` and `rightRng`). We always merge the right-hand side range into the left-hand side range.